netCDF4/utils.py

```diff
@@ -269,7 +269,7 @@ def _StartCountStride(elem, shape, dimensions=None, grp=None, datashape=None,\
                 ee = range(start,stop,step)
             except ValueError: # start, stop or step is not valid for a range
                 ee = False
-            if ee and len(e) == len(ee) and (e == np.arange(start,stop,step)).all():
+            if ee and e[1]-e[0]==1 and len(e) == len(ee) and (e == np.arange(start,stop,step)).all():
```

Comment:
a step of -1 should be convert to a slice also, so the check should be `step not in [1,-1]`